inttestutils/transfer.go

```diff
@@ -0,0 +1,240 @@
+	coreutils.ExitOnErr(targetArtifactoryCli.Exec("repo-delete", tests.MvnRemoteRepo))
+}
+
+// Clean test repositories content in the target Artifactory
+// targetArtifactoryCli - Target Artifactory CLI
+func CleanTargetRepos(targetArtifactoryCli *tests.JfrogCli) {
+	coreutils.ExitOnErr(targetArtifactoryCli.Exec("del", tests.RtRepo1))
+	coreutils.ExitOnErr(targetArtifactoryCli.Exec("del", tests.RtRepo2))
+	coreutils.ExitOnErr(targetArtifactoryCli.Exec("del", tests.MvnRepo1))
+	coreutils.ExitOnErr(targetArtifactoryCli.Exec("del", tests.MvnRemoteRepo))
+}
+
+// Install data-transfer Artifactory user plugin
+func InstallDataTransferPlugin() {
+	pluginsDir := filepath.Join(*tests.JfrogHome, "artifactory", "var", "etc", "artifactory", "plugins")
+	groovyFile := filepath.Join(pluginsDir, groovyFileName)
+	err := buildInfoGoUtils.DownloadFile(groovyFile, dataTransferUrl+groovyFileName)
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+
+	libDir := filepath.Join(pluginsDir, "lib")
+	fileutils.CreateDirIfNotExist(libDir)
+	jarFile := filepath.Join(libDir, jarFileName)
+	err = buildInfoGoUtils.DownloadFile(jarFile, dataTransferUrl+"lib/"+jarFileName)
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+}
+
+// Authenticate target Artifactory using the input flags
+func AuthenticateTarget() (string, *config.ServerDetails) {
+	*tests.JfrogTargetUrl = clientutils.AddTrailingSlashIfNeeded(*tests.JfrogTargetUrl)
+	serverDetails := &config.ServerDetails{
+		Url:            *tests.JfrogTargetUrl,
+		ArtifactoryUrl: *tests.JfrogTargetUrl + tests.ArtifactoryEndpoint,
+		AccessToken:    *tests.JfrogTargetAccessToken,
+	}
+	cred := "--url=" + serverDetails.ArtifactoryUrl + " --access-token=" + serverDetails.AccessToken
+	if _, err := serverDetails.CreateArtAuthConfig(); err != nil {
+		coreutils.ExitOnErr(errors.New("Failed while attempting to authenticate with Artifactory: " + err.Error()))
+	}
+	return cred, serverDetails
+}
+
+// Verify the input slice exist in the target Artifactory
+// expected - The slice to check
+// specFile - File spec for the search command
+// serverDetails - Target Artifactory server details
+// t - Tests object
+func VerifyExistInTargetArtifactory(expected []string, specFile string, serverDetails *config.ServerDetails, t *testing.T) {
+	results, _ := SearchInArtifactory(specFile, serverDetails, t)
+	tests.CompareExpectedVsActual(expected, results, t)
+}
+
+// Refresh the storage info and wait.
+// This is required to allow the next /storageInfo REST call to return the repositories created in the test.
+// serverDetails - Source Artifactory server details
+func RefreshStorageInfoAndWait(sourceServerDetails *config.ServerDetails) error {
+	// Create service manager
+	serviceManager, err := utils.CreateServiceManager(sourceServerDetails, -1, 0, false)
+	if err != nil {
+		return err
+	}
+
+	// Refresh and get storage info with refresh=true.
+	storageInfo, err := serviceManager.StorageInfo(true)
```

Comment:
Since the refresh is async, this makes this REST require the following functionality, which makes it pretty useless. WDYT? We should maybe split it to two separate apis.

Reply:
I agree. We should talk about it and decide how to make it right.